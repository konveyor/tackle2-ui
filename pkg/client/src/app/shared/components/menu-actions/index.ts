export { MenuActions } from "./menu-actions";
