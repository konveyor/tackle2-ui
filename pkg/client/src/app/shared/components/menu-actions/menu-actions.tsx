import React, { useState } from "react";
import { Dropdown, DropdownItem, DropdownToggle } from "@patternfly/react-core";
import { CaretDownIcon } from "@patternfly/react-icons/dist/esm/icons/caret-down-icon";

export interface MenuActionsProps {
  actions: { label: string; callback: () => void }[];
}

export const MenuActions: React.FC<MenuActionsProps> = ({ actions }) => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <Dropdown
      isOpen={isOpen}
      onSelect={() => {
        setIsOpen(!isOpen);
      }}
      toggle={
        <DropdownToggle
          onToggle={(isOpen: boolean) => {
            setIsOpen(isOpen);
          }}
          toggleIndicator={CaretDownIcon}
        >
          Actions
        </DropdownToggle>
      }
      dropdownItems={actions.map((element, index) => (
        <DropdownItem key={index} component="button" onClick={element.callback}>
          {element.label}
        </DropdownItem>
      ))}
    />
  );
};
