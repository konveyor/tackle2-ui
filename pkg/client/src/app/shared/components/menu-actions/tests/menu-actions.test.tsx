import React from "react";
import { mount, shallow } from "enzyme";
import { MenuActions } from "../menu-actions";

describe("MenuActions", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <MenuActions
        actions={[
          { label: "Action1", callback: jest.fn() },
          { label: "Action2", callback: jest.fn() },
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Executes callback", () => {
    const callback1Mock = jest.fn();
    const callback2Mock = jest.fn();

    const wrapper = mount(
      <MenuActions
        actions={[
          { label: "Action1", callback: callback1Mock },
          { label: "Action2", callback: callback2Mock },
        ]}
      />
    );

    // Select dropdown btn
    const dropdownBtn = wrapper.find("button").at(0);
    expect(dropdownBtn.text()).toEqual("Actions");

    // Verify callbacks are executed

    dropdownBtn.simulate("click"); // Opens dropdown

    const action1Btn = wrapper.find(".pf-c-dropdown__menu-item").at(0);
    expect(action1Btn.text()).toEqual("Action1");
    action1Btn.simulate("click");
    expect(callback1Mock).toHaveBeenCalledTimes(1);

    dropdownBtn.simulate("click"); // Opens dropdown
    const action2Btn = wrapper.find(".pf-c-dropdown__menu-item").at(1);
    expect(action2Btn.text()).toEqual("Action2");
    action2Btn.simulate("click");
    expect(callback2Mock).toHaveBeenCalledTimes(1);
  });
});
