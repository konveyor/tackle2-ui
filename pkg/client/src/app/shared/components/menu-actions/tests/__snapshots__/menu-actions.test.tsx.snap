// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuActions Renders without crashing 1`] = `
<Dropdown
  dropdownItems={
    Array [
      <DropdownItem
        component="button"
        onClick={[MockFunction]}
      >
        Action1
      </DropdownItem>,
      <DropdownItem
        component="button"
        onClick={[MockFunction]}
      >
        Action2
      </DropdownItem>,
    ]
  }
  isOpen={false}
  onSelect={[Function]}
  toggle={
    <DropdownToggle
      onToggle={[Function]}
    >
      Actions
    </DropdownToggle>
  }
/>
`;
